c/src/CMakeLists.txt

```diff
@@ -16,9 +18,25 @@ swift_add_library(sbp
 )
 add_library(swiftnav::sbp ALIAS sbp)
 
-target_include_directories(sbp PUBLIC ${PROJECT_SOURCE_DIR}/include PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
-set_target_properties(sbp PROPERTIES
-        POSITION_INDEPENDENT_CODE ON)
+generate_export_header(sbp
+  DEPRECATED_MACRO_NAME CMAKE_SBP_DEPRECATED
```

Comment:
these `*DEPRECATED_MACRO_NAME` entries needed to be added because the `generate_export_header` otherwise defines `SBP_DEPRECATED` which conflicts with the `SBP_DEPRECATED` definition in `common.h`.

Also the `INCLUDE_GUARD_NAME` is to make sure that it complies with the Google C++ Style Guide

Reply:
Wouldn't it be simpler to just add `SBP_EXPORT` to `common.h` along with all the other compiler macros? That's the way [libswiftnav handles this same problem](https://github.com/swift-nav/libswiftnav-private/blob/3f9334a4779135e91a433ddf18a3c8a90c388456/include/swiftnav/macros.h#L132). Then all the compiler specific stuff can be in one place and we don't have to deal with duplicate symbols

Reply:
`generate_export_header` manages things across all compilers nicely for you without any extra work and or knowledge, the naming clash is unfortunate for `SBP_DEPRECATED`, but not a major issue. Also I don't know if the feature even works on `libswiftnav` since `SWIFT_DECLSPEC` is only being used by two methods and no CI stage to build shared libraries on Windows OS.

The use of this is also advised by the cmake maintainer as he said that people have tried to write this on their own for all platforms, but often have issues so it is better to stick with this approach.

Reply:
Yeah I understand, but we have to deal with this issue anyway since `generate_export_header` is very limited in scope - we also have to have macros for packed, format, etc. From a maintainability perspective I think it's better to have all of this sort of stuff in the same place rather than having 2 different mechanism for defining compiler attribute macros.

Reply:
Made the changes as requested, will also see if I can add a test to catch the installation issue you posted below.